src/include/switch_rtp.h

```diff
@@ -57,6 +57,7 @@ typedef struct {
 	char body[SWITCH_RTP_MAX_BUF_LEN+4+sizeof(char *)];
 	switch_rtp_hdr_ext_t *ext;
 	char *ebody;
+	switch_time_t received_ts;
```

Comment:
This should not be here, nor in the rtp_msg_t . If you want to to add the received timestamp to the JB you will have to create a different struct with both switch_rtp_packet_t and switch_time_t in it.
Expose it in switch_rtp.h and use if for switch_jb_put_packet() and switch_jb_get_packet() (at least) .
eg:
typedef struct {
        switch_rtp_packet_t pkt;
        switch_time_t rcvd_time;
} switch_rtp_incoming_pkt_t;

Reply:
I see your point, I wanted to measure from the input to the output socket to be sure that the measurements are taken as close as possible to the input and output.

Do I understand that adding this at the packet level is to intrusive and measuring only at the jitter buffer in/out would be preferable ?

This approach would probably require a bit more code modification since some functions parameters will 
have to be modified to support this new struct.

Thanks for the feedback i will take some time to consider and look at it.

Reply:
you'll need a rcvd_time in the switch_jb_node_t too.

Reply:
It's the sizeof() of those structs (rtp_msg_t or switch_rtp_packet_t) - better to keep them as they are in master branch now.